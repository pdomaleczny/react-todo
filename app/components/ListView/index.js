import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Delete from 'material-ui/svg-icons/action/delete';
import { List, ListItem } from 'material-ui/List';
import Subheader from 'material-ui/Subheader';

class ListView extends Component {
  renderTitle() {
    return this.props.tasks.length > 0 ? 'My Tasks ðŸ—ƒ' : 'You have no tasks. Add one ðŸ‘Œ';
  }

  deleteTask(taskIndex) {
    this.props.deleteTask(taskIndex);
  }

  render() {
    return (
      <List>
        <Subheader>{this.renderTitle()}</Subheader>
        {
          this.props.tasks.map((el, index) =>
            <ListItem
              key={index}
              primaryText={el}
              rightIcon={<Delete onTouchTap={() => { this.deleteTask(index); }} />}
            />,
          )
        }
      </List>
    );
  }
}

ListView.propTypes = {
  tasks: PropTypes.arrayOf(PropTypes.string),
  deleteTask: PropTypes.func.isRequired,
};

export default ListView;
