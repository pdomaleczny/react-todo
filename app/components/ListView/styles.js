export const headerStyles = {
  fontSize: 42,
  textAlign: 'center',
  color: '#000',
  paddingBottom: 20,
  marginBottom: 20,
  fontWeight: 300,
};

export default {
  headerStyles,
};
