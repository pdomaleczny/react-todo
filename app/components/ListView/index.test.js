/* eslint-disable */
import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { shallow } from 'enzyme';
import ListView from '../ListView';
import Subheader from 'material-ui/Subheader';
import Delete from 'material-ui/svg-icons/action/delete';

it('should render a ListView', () => {
    const tasks = ['Test', 'Test2'];
    const wrapper = shallow(
        <ListView tasks={tasks}/>
    );
    expect(wrapper).toMatchSnapshot();
});

it('should pass tasks as props', () => {
    const tasks = ['Test', 'Test2'];
    const deleteTaskMock = jest.fn();
    const wrapper = shallow(
        <ListView tasks={tasks} deleteTask={(taskIndex) => { deleteTaskMock(); }}/>
    );
    expect(wrapper.find('ListItem').length).toEqual(2);
    expect(wrapper.find(Subheader).props().children).toEqual('My Tasks ðŸ—ƒ');
});

it('should return different title when no tasks', () => {
    const tasks = [];
    const deleteTaskMock = jest.fn();
    const wrapper = shallow(
        <ListView tasks={tasks} deleteTask={(taskIndex) => { deleteTaskMock(); }}/>
    );
    expect(wrapper.find('ListItem').length).toEqual(0);
    expect(wrapper.find(Subheader).props().children).toEqual('You have no tasks. Add one ðŸ‘Œ');
});

it('should trigger delete action', () => {
    const tasks = ['Test', 'Test2'];
    const wrapper = shallow(
        <ListView tasks={tasks}/>
    );

    wrapper.instance().deleteTask = jest.fn();
    wrapper.update();

    wrapper.find(Delete).simulate('touchTap');

    expect(wrapper.instance().deleteTask.mock.calls.length).toEqual(1);
});

