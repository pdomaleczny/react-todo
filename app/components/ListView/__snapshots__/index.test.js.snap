// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a ListView 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <List>
    <Subheader
        inset={false}
        style={
            Object {
                "color": "#000",
                "fontSize": 42,
                "fontWeight": 300,
                "marginBottom": 20,
                "paddingBottom": 20,
                "textAlign": "center",
              }
        }
    >
        My Tasks ðŸ—ƒ
    </Subheader>
    <ListItem
        autoGenerateNestedIndicator={true}
        containerElement="span"
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="Test"
        primaryTogglesNestedList={false}
        rightIcon={
            <ActionDelete
                onTouchTap={[Function]}
            />
        }
        secondaryTextLines={1}
    />
    <ListItem
        autoGenerateNestedIndicator={true}
        containerElement="span"
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="Test2"
        primaryTogglesNestedList={false}
        rightIcon={
            <ActionDelete
                onTouchTap={[Function]}
            />
        }
        secondaryTextLines={1}
    />
</List>,
  "nodes": Array [
    <List>
      <Subheader
            inset={false}
            style={
                  Object {
                        "color": "#000",
                        "fontSize": 42,
                        "fontWeight": 300,
                        "marginBottom": 20,
                        "paddingBottom": 20,
                        "textAlign": "center",
                      }
            }
      >
            My Tasks ðŸ—ƒ
      </Subheader>
      <ListItem
            autoGenerateNestedIndicator={true}
            containerElement="span"
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="Test"
            primaryTogglesNestedList={false}
            rightIcon={
                  <ActionDelete
                        onTouchTap={[Function]}
                  />
            }
            secondaryTextLines={1}
      />
      <ListItem
            autoGenerateNestedIndicator={true}
            containerElement="span"
            disableKeyboardFocus={false}
            disabled={false}
            initiallyOpen={false}
            insetChildren={false}
            nestedItems={Array []}
            nestedLevel={0}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onNestedListToggle={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            open={null}
            primaryText="Test2"
            primaryTogglesNestedList={false}
            rightIcon={
                  <ActionDelete
                        onTouchTap={[Function]}
                  />
            }
            secondaryTextLines={1}
      />
</List>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ListView
        tasks={
                Array [
                        "Test",
                        "Test2",
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ListView {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "tasks": Array [
            "Test",
            "Test2",
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <List>
          <Subheader
                    inset={false}
                    style={
                              Object {
                                        "color": "#000",
                                        "fontSize": 42,
                                        "fontWeight": 300,
                                        "marginBottom": 20,
                                        "paddingBottom": 20,
                                        "textAlign": "center",
                                      }
                    }
          >
                    My Tasks ðŸ—ƒ
          </Subheader>
          <ListItem
                    autoGenerateNestedIndicator={true}
                    containerElement="span"
                    disableKeyboardFocus={false}
                    disabled={false}
                    initiallyOpen={false}
                    insetChildren={false}
                    nestedItems={Array []}
                    nestedLevel={0}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onNestedListToggle={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    open={null}
                    primaryText="Test"
                    primaryTogglesNestedList={false}
                    rightIcon={
                              <ActionDelete
                                        onTouchTap={[Function]}
                              />
                    }
                    secondaryTextLines={1}
          />
          <ListItem
                    autoGenerateNestedIndicator={true}
                    containerElement="span"
                    disableKeyboardFocus={false}
                    disabled={false}
                    initiallyOpen={false}
                    insetChildren={false}
                    nestedItems={Array []}
                    nestedLevel={0}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onNestedListToggle={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    open={null}
                    primaryText="Test2"
                    primaryTogglesNestedList={false}
                    rightIcon={
                              <ActionDelete
                                        onTouchTap={[Function]}
                              />
                    }
                    secondaryTextLines={1}
          />
</List>,
        "_debugID": 2,
        "_renderedOutput": <List>
          <Subheader
                    inset={false}
                    style={
                              Object {
                                        "color": "#000",
                                        "fontSize": 42,
                                        "fontWeight": 300,
                                        "marginBottom": 20,
                                        "paddingBottom": 20,
                                        "textAlign": "center",
                                      }
                    }
          >
                    My Tasks ðŸ—ƒ
          </Subheader>
          <ListItem
                    autoGenerateNestedIndicator={true}
                    containerElement="span"
                    disableKeyboardFocus={false}
                    disabled={false}
                    initiallyOpen={false}
                    insetChildren={false}
                    nestedItems={Array []}
                    nestedLevel={0}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onNestedListToggle={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    open={null}
                    primaryText="Test"
                    primaryTogglesNestedList={false}
                    rightIcon={
                              <ActionDelete
                                        onTouchTap={[Function]}
                              />
                    }
                    secondaryTextLines={1}
          />
          <ListItem
                    autoGenerateNestedIndicator={true}
                    containerElement="span"
                    disableKeyboardFocus={false}
                    disabled={false}
                    initiallyOpen={false}
                    insetChildren={false}
                    nestedItems={Array []}
                    nestedLevel={0}
                    onKeyboardFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onNestedListToggle={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    open={null}
                    primaryText="Test2"
                    primaryTogglesNestedList={false}
                    rightIcon={
                              <ActionDelete
                                        onTouchTap={[Function]}
                              />
                    }
                    secondaryTextLines={1}
          />
</List>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ListView
    tasks={
        Array [
            "Test",
            "Test2",
          ]
    }
/>,
}
`;
